debian/setup-user.sh

```diff
@@ -6,3 +6,5 @@ adduser --gecos user \
   user
 adduser user sudo
 echo "user:user" | chpasswd
+sudo adduser --system ggc_user
```

Comment:
@lukasz-laguna I am not sure if we should do that in the image. The `Greengrass` is only one of the usecases of the system. Let's remove the GGC specific bits from there:
- user and group creation,
- sqlite3

We will instruct in the documentation how to install those.

Hard/soft link protection is a general improvement and we can leave that.